site/source/tutorials/1030_routing/index.md

```diff
@@ -0,0 +1,183 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/outlets/WorkerContainerOutlet.ts' %}
+
+{% instruction 'Add the follow code to `BannerOutlet.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/outlets/BannerOutlet.ts' %}
+
+Now, we can the swap out the widgets for the newly created `outlets`. Notice we have not needed to changed any of wrapped widgets!
+
+{% instruction 'First, swap the widget imports for the outlet imports in `App.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:8-10 %}
+
+{% instruction 'Then, swap the widget usages for their outlets.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:50-55,59-61 %}
+
+Running the application now should display the `Banner.ts` by default, but also enable routing to the other widgets using the `/#directory` and `/#new-widget` routes.
+
+Next, we will add a side menu with links for the created outlets.
+
+{% section %}
+
+## Adding Links
+
+{% task 'Add a sidebar menu to the application.' %}
+
+In this section we will be using the `Link` component, provided by `@dojo/routing`, to create link elements with an `href` for an outlet name. A `label` for the `Link` can be passed as children and `param` values for the outlet can be passed to a `Link` component using the `params` property.
+
+```ts
+w(Link, { to: 'outlet-name', params: { paramName: 'value' } });
+
+```
+
+{% instruction 'Add the `Link` import in `App.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' line:4 %}
+
+{% instruction 'Replace the render function in `App.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/App.ts' lines:37-55,59-64 %}
+
+Now, the links in the side menu can be used to navigate around the application!
+
+{% section %}
+
+## Dynamic `Outlet`
+
+{% task 'Add filters to the `WorkerContainer.ts` widget and create an outlet.' %}
+
+Finally, we are going to enhance the `WidgetContainer.ts` with a filter on the workers' last name. To do this we need to use the `filter` outlet configured in the first section. The key difference for the `filter` outlet is that the path is using a placeholder that indicates a path parameter, `{filter}`.
+
+This means a route with any value will match the `filter` as long as the previous path segments match, so for the `filter` outlet a route of `directory/any-value-here` would be considered a match.
+
+{% instruction 'Add the new property to the `WidgetContainerProperties` interface in `WorkerContainer.ts`.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' lines:10-13 %}
+
+{% instruction 'Include the `Link` import' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' line:8 %}
+
+{% instruction 'Add a private function to generate the filter links' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/widgets/WorkerContainer.ts' lines:20-32 %}
```

Comment:
Missing import for `DNode` used in this function

Reply:
thought I'd removed the type (which I have now 😄 )